### PR TITLE
Remove redundant null checks for atmos

### PR DESCRIPTION
**Pull request by DrSmugleaf**

### CODE REVIEW

File: Content.Server/Atmos/GasMixture.cs [hunk 1]
```diff
@@ -146,7 +145,7 @@ public void Archive()
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void Merge(GasMixture giver)
+        public void Merge(GasMixture? giver)
         {
             if (Immutable || giver == null) return;
```
Comment (PJB3005) [inline Content.Server/Atmos/GasMixture.cs:150]: Remove this `giver == null` then?

Comment (DrSmugleaf) [reply Content.Server/Atmos/GasMixture.cs:150]: This is adding nullability to the `giver` argument though so if I do that it's going to cause a compile error in 152 when accessing giver, or should I remove both the nullability and the null check?

Comment (Zumorica) [reply Content.Server/Atmos/GasMixture.cs:150]: Yeah, `giver` should never be null so I guess removing both the check and nullability is what should be done here

Comment (PJB3005) [reply Content.Server/Atmos/GasMixture.cs]: oh yeah I misread the diff, my bad.

**Review by Zumorica — APPROVED** [review]

**Pull request merged**